proto/vschema.proto

```diff
@@ -64,7 +64,11 @@ message Table {
   // shard, as dictated by the keyspace id.
   // The keyspace id is represened in hex form
   // like in keyranges.
-  string pinned =5;
+  string pinned = 5;
+  // is_authoritative is set ot true if columns is
```

Comment:
It seems to me that `is_authoritative` is too broad of an attribute name for this feature.
After all the vschema is always authoritative for the sharding controls, etc.

How about we keep this more focused and call the attribute "authoritative_columns" or something of that sort.

Also typo "ot" above.

Reply:
This is a calculated risk. My prediction is that the next step will be to pull the schema info from vttablet, at which time the meaning of this flag will become more accurate.

If you think it will backfire, I'll change it.

Reply:
IMO this potential future is all the more reason to use a more targeted configuration now that takes into account future expansions. Specifically, once the schema is propagated from the vttablet, I'd expect that by default we wouldn't want to set this flag and that instead all information would be pulled from the tablets.

What if we replaced the `is_authoritative` boolean with a `schema_source` enum containing options:

 * `DEFAULT` -- legacy behavior, eventually to be enhanced with schema info pulled from vttablet
 * `STATIC` -- the new behavior being added in this PR
 * `NONE` -- same as legacy behavior for now, but will disable the feature where schemas are pulled from vttablet.

If we go down this road we have a decision to make around the weight_string sorting changes made in #3522. IMO a consistent plan would be to couple the enabling of those changes with the changes in this PR and make it so that by default the `columns` definition in the vschema is ignored, but that if the `schema_source` is `static` then we enable both the weight_string replacement and the expanded column list for `select *` changes in this PR. 

That approach would, however, break compatibility for anyone who is dependent on the current weight_string implementation and uses `select *` (including Slack FWIW) since users would need to be sure to fill in _all_ the column names into the vschema and enable the new feature.

The other approach would be to just control the existing feature with the new flags and leave the weight_string sorting alone. This is better for backwards compatibility but would be less internally consistent.

Reply:
As far as I can tell, #3522 and this feature are fairly orthogonal. Column expansion happens early, at the time of select expression parsing, and `weight_string` decision is made later during `order by` processing. So, at that time, the columns will already be expanded out, which will cause `weight_string` to be added if needed.

I'll add a test case to demonstrate this behavior, which will also help me prove to myself that this works as intended.

As for `is_authoritative`: you seem to think it's not appropriate. I'm fine changing it to `authoritative_columns`.

Reply:
The reason I think the two are linked isn't anything related to the code that does the expansion, it's that weight_string ordering and select * expansion both depend on the column definitions being manually populated in the vschema.

So it seems to me that we should take both features into account when we start talking about ways to control whether or not vtgate should respect the column definitions in the vschema to inform the query planner or whether it should operate as best it can without knowing the schema.

Reply:
I'm ok to go ahead with `authoritative_column_list` instead of the enum approach outlined above. We may need to revisit in the future assuming we pull the schema list from vttablet.